snooty/postprocess.py

```diff
@@ -170,6 +170,7 @@ def run(
     def generate_metadata(self) -> n.SerializedNode:
         document: Dict[str, SerializableType] = {}
         document["title"] = self.project_config.title
+        document["deprecated_versions"] = self.project_config.deprecated_versions
```

Comment:
When I parsed a repo that doesn't have a `deprecated_versions` table in its snooty.toml, the `metadata` document contained the line `deprecated_versions: null`. We need to wrap this in a conditional so that it's only included if that field is defined:
```python
if self.project_config.deprecated_versions:
    document["deprecated_versions"] = self.project_config.deprecated_versions
```